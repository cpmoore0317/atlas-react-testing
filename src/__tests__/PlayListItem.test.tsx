import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';
import { expect, it, describe } from 'vitest';

describe('PlayListItem Component', () => {
    it('renders correctly when selected', () => {
        const { container } = render(
            <PlayListItem
                title='Test Song'
                artist='Test Artist'
                duration='3:30'
                isSelected={true}
                onClick={() => {}}
            />    
        );
        expect(container).toMatchSnapshot();
    });

    it('renders correctly when not selected', () => {
        const { container } = render(
            <PlayListItem
                title="Another Song"
                artist="Another Artist"
                duration="4:00"
                isSelected={false}
                onClick={() => {}}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
