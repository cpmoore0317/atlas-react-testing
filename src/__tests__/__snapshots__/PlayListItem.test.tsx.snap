// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component > renders correctly when not selected 1`] = `
<div>
  <div
    class="flex justify-between items-center py-2 "
  >
    <div
      class="flex-grow w-32"
    >
      <h3
        class="text-sm font-bold ml-1"
      >
        Another Song
      </h3>
      <p
        class="text-xs text-gray-500 ml-1"
      >
        Another Artist
      </p>
    </div>
    <span
      class="text-xs text-gray-500 text-right ml-72 mr-1"
    >
      4:00
    </span>
  </div>
</div>
`;

exports[`PlayListItem Component > renders correctly when selected 1`] = `
<div>
  <div
    class="flex justify-between items-center py-2 bg-gray-300"
  >
    <div
      class="flex-grow w-32"
    >
      <h3
        class="text-sm font-bold ml-1"
      >
        Test Song
      </h3>
      <p
        class="text-xs text-gray-500 ml-1"
      >
        Test Artist
      </p>
    </div>
    <span
      class="text-xs text-gray-500 text-right ml-72 mr-1"
    >
      3:30
    </span>
  </div>
</div>
`;
